app/assets/javascripts/hubstats/application.js

```diff
@@ -14,45 +14,85 @@
 //= require jquery_ujs
 //= require select2
 //= require hubstats/bootstrap
+//= require bootstrap-datepicker
 //= require_tree .
 
 $(document).ready( function() {
+  $('.input-daterange').datepicker({
+    "todayHighlight": true,
+    "endDate": "Today",
+    "todayBtn": true
+  });
   setDateRange()
 });
 
 function setDateRange() {
-  var index = readCookie("hubstats_index") || 2;
-  var timer = document.getElementById("time-select");
+  var dates;
+  var timer;
+  var datesArray = [];
 
-  timer.selectedIndex = index;
+  if ((readCookie("hubstats_dates") === "null~~null") || (readCookie("hubstats_dates") === null)) {
+    dates = getDefaultDateRange();
+  } else {
+    dates = readCookie("hubstats_dates")
+  }
 
-  timer.onchange = function() {
-    createCookie("hubstats_index",this.selectedIndex,1);
+  timer = document.getElementById("submitDateRange");
+  datesArray = dates.split("~~")
+
+  $('.input-daterange').find('[name="start"]').datepicker('update', new Date(datesArray[0]));
```

Comment:
Save these elements to a variable and re-use them later on to avoid parsing the DOM over again.  When a page gets large using anything but an ID can be an expensive operation.
```
var startInput = $('.input-daterange [name="start"]');
var endInput = $('.input-daterange [name="end"]');
startInput.datepicker(...
```